### DISCUSSION TITLE
Edge

### DISCUSSION THREAD

**Discussion by Jigoku** [redeclipse/base#1198] [Maps and Content Showcase]
### Edge 
A cyberpunk/industrial sci-fi themed map for RE2.

----

![20200105020009](https://user-images.githubusercontent.com/1535179/71774023-50aaf080-2f5f-11ea-9b34-127de7f02627.png)
![20200105020016](https://user-images.githubusercontent.com/1535179/71774025-543e7780-2f5f-11ea-9a91-3a2b3fde1e23.png)
![20200105020028](https://user-images.githubusercontent.com/1535179/71774026-56a0d180-2f5f-11ea-80c3-0d8a3632c9ee.png)

----

Ideal gameplay:
* ffa / duel / survivor / bomberball-hold
* 2-8 players (seems to work with up to 10-15 at most)

The map also supports team spawns, along with dac/ctf/bomber on the side, i have no idea how it will place in those modes (seems alright with bot matches), have also set `mapbalance 1` as for the asymmetrical layout. It's probably best suited as a ffa map however.

### Pull requests open here:
redeclipse/maps#59
redeclipse/misc#2 

It's worth pointing out, i've tried my best to make sure almost every surface is usable with the parkour mechanics, there are lots of ways to get around the map very quickly that might not be too obvious at first, so hopefully there's a bit of fun for advanced parkour movement without it becoming cumbersome to newer players. The general layout is relatively simple, based around an 8 shape on an incline, but also has areas above the lower rooftops, and some passages to cut through and over.

**Comment by Jigoku** [comment]
Updated download link to r1432 (was missing fxlevelled mapmodels)

**Comment by Calinou** [comment]
I haven't tested it yet, but it looks very nice. Great work :slightly_smiling_face:

**Comment by cloudziq** [comment]
Map with good texturing, usage of new features, without confusing layout and what is more important - the first ever map for 2.0 that looks really good, with a nice consistent theme around it.
I love the look and feel of this map, map which can serve as a 'quality example' for other authors in my opinion

**Comment by Jigoku** [comment]
**Updated to r1571**

* texture/clip fixes
* 2 new decals for some varied adverts/billboards
* retexture/detail on most of the buildings
* skytex optimizations on map bounds
* more fxlevels for some models (noshadow etc)
* added player spawns to rooftop areas
* swapped one of the zapper / shotgun spawns about
* some tweaks to vcolour texturing

Frame rates should be more stable now, as i've sorted out some of the lighting issues that were degrading performance.

There's still a bit more texture/geometry work to do, mostly inside the "loading bay" room, and next to that soup decal with the curved lit windows. And possibly 1 or 2 more new textures that can function as billboards/adverts of sorts.

----
![20200105020113](https://user-images.githubusercontent.com/1535179/71774077-19890f00-2f60-11ea-8b38-6c27ebe6a7f2.png)
![20200105020107](https://user-images.githubusercontent.com/1535179/71774078-1c83ff80-2f60-11ea-8295-cc360e906a8b.png)

**Comment by enp2s0** [comment]
What's the ETA for the final map? I'd be interested in adding it to my server even if it isn't included in the official game.

**Comment by no-lex** [comment]
This map is, as indicated on the sidebar, planned for 2.0.1 which will release in a few months.

**Comment by Jigoku** [comment]
> What's the ETA for the final map? I'd be interested in adding it to my server even if it isn't included in the official game.

You can download the zip now and put it on a server if you want, but be warned there are some extra assets in the *unixfreak/edge/* folder, anyone who connects / getmap will see several texture errors as the data files aren't sent across servers (only the mpz,cfg,txt,wpt files are)

**Comment by Jigoku** [comment]
**Updated to r1658**

* texture/clip fixes
* waypoints adjusted
* small lighting/ambient tweaks for both variants
* some floor decals rescaled
* removed empty cubes
* changed 2 of the large building models
* added affinities/spawns for ctf/dac/team gameplay
* mapbalance = 1

![20200109080846](https://user-images.githubusercontent.com/1535179/72049702-66018100-32b7-11ea-9349-7a895119d402.png)
![20200109080926](https://user-images.githubusercontent.com/1535179/72049704-69950800-32b7-11ea-9199-ef062b11b8c7.png)

**Comment by Jigoku** [comment]
**Updated to r1703**
* texture fixes
* clip fixes
* small light tweaks

**Comment by dc-redeclipse** [comment]
you exclude the cfg file in your zip. I tryed without and see flying trees ;D

**Comment by Jigoku** [comment]
> you exclude the cfg file in your zip. I tryed without and see flying trees ;D

oops, you're right, i'll update again (slightly more recent version)

**Comment by Jigoku** [comment]
**Updated to r1714**

* misc texture fixes 
* texturecull
* re-add missing cfg

**Comment by dc-redeclipse** [comment]
Nice design! good work but is this high shooting point wanted?
![20200118195051](https://user-images.githubusercontent.com/60038236/72668931-23b11000-3a2c-11ea-8ea6-1a2d45c40e1e.png)

**Comment by Jigoku** [comment]
> Nice design! good work but is this high shooting point wanted?

Thanks.Didn't notice that was possible to stand in, it should be fixed yeah. I think it's a problem with the skytextured cubes i'm using to add clipping along that angled edge. They have a small gap below as to not occlude neighbouring textures.

I will see if i can replace those cubes later, with alpha material (using the `valpha 0` trick, so they're invisible).

**Comment by dc-redeclipse** [comment]
Maybe not a bug but is this texture offset correct?
![20200118200730](https://user-images.githubusercontent.com/60038236/72669173-00d42b00-3a2f-11ea-9df7-77e8dc90dee4.png)

I tried an offset in y of 321 an it looked way more centered then before.  
![20200118201051](https://user-images.githubusercontent.com/60038236/72669172-00d42b00-3a2f-11ea-981b-14eb5136207e.png)
But I don't know if that is wanted or not.

**Comment by Jigoku** [comment]
Yeah that texture is intended to be offset like that.

**Comment by Jigoku** [comment]
**Updated to r1735**

* texture / material fixes
* use misc/panel_blank for holosphere mapmodel instead of 1x1 white pixel
* redo tvmount mapmodel mesh/texture/normals

Can't spot any obvious problems with the map right now. 
So @qreeves if you wanted to take a look at it, i consider it finished for now :stuck_out_tongue:

**Comment by Jigoku** [comment]
**Updated to r1737**

* added affinity for $mutsbitgsp2 / KOTH mutator at the centre of the map

**Comment by Jigoku** [comment]
**Updated to r1741**

* added point names / ids for dac capture points

Forgot to do this earlier;

```
point_4 = [ Central Square ]
point_2 = [ Alpha Base     ]
point_3 = [ Loading Bay    ]
point_1 = [ Omega Base     ]
```

Should be last update to this map for now, unless any bugs/clip problems are found.

**Comment by dc-redeclipse** [comment]
![20200121181143](https://user-images.githubusercontent.com/60038236/72826667-8d722980-3c79-11ea-8d21-09a5d81d5913.png)
Is that propeller missing?

**Comment by Jigoku** [comment]
**Updated to r1758**

* re-added missing propeller on a jumppad (thanks dc-de)
* several geometry fixes (player getting stuck)
* some small blend attr tweaks on some decals
* replaced blue flat texture above the cyber soup area with a scrolling pattern

**Comment by Jigoku** [comment]
**Updated to r1763**

* add static animation to tvmount models
* reduce tvmount texture/normal filesizes

**Comment by Jigoku** [comment]
**Updated to r1768**

* several tiny texture fixes

Combed through the entire map the past hour, can't spot any more issues, will open a pull request shortly.

**Comment by Jigoku** [comment]
**Updated to r1791**

* re-add a missing background building model
* move envmap decal/puddles from fxlevel3 -> fxlevel2
* small blendmap fixes where two envmap entities overlap
* add stair lights to guns/ammo building
* fix stair surfaces in a few places

**Comment by Jigoku** [comment]
**Updated to r1795**

* added 2 sword spawns for classic mutator

**Comment by fkorax** [comment]
Awesome map! 💯 
Teams should definitely be reassigned for map symmetry/balance if this is to be included;
Alpha has a clear advantage over Omega in the current setup.

**Comment by Jigoku** [comment]
@thomorl cheers :+1:  
yeah due to the unsymmetrical/abstract layout it's intended to be a ffa map, although it does have `mapbalance 1` set, which forces a team swap halfway for ctf / dac / bb modes if they are picked.

**Comment by fkorax** [comment]
@Jigoku I would set `mapbalance 1` in deathmatch/team mode as well. I played the map in that mode for a while now, and Omega wins virtually every time.

**Comment by Jigoku** [comment]
Hmm, i think mapbalance should apply to all modes

**Comment by Jigoku** [comment]
Oops, ignore my last comment, `mapbalance 2` is what was needed as to apply to all modes. I'll make a PR now, thanks for pointing that out :+1:
